include/rocksdb/statistics.h

```diff
@@ -188,12 +188,6 @@ enum Tickers : uint32_t {
   // Record the number of calls to GetUpdatesSince. Useful to keep track of
   // transaction log iterator refreshes
   GET_UPDATES_SINCE_CALLS,
-  BLOCK_CACHE_COMPRESSED_MISS,  // miss in the compressed block cache
```

Comment:
I'm not sure of the consequences of re-numbering existing statistics. @ltamasi

Reply:
Right, enum values on the C++ and Java sides have to be kept in sync.

Reply:
Hm, actually on closer inspection, the ids are already out of sync in various ways. Seems we first diverge at `BLOOM_FILTER_FULL_POSITIVE`, which would be 21 == 0x15 in the C++ code but missing from the JNI (where 0x15 is actually `PERSISTENT_CACHE_HIT`). Probably stuff got added in the middle at some point.

Reply:
Too bad I just spent half an hour changing the numbers and filled the gap...

Reply:
It definitely would have been nice if the values had been kept in sync but that apparently that ship has sailed; even if we fill this gap, there are pre-existing tickers which have different ids between C++ and Java (e.g. `BLOOM_FILTER_FULL_POSITIVE` is 0x63 on the Java side).

There's also this comment in `portal.h`; I'm not sure what the original rationale behind exposing `TICKER_ENUM_MAX` was (especially since its value is guaranteed to change in the core library whenever a ticker is added) but seems like the convention in the Java bindings is to keep the ids the same:

```
      case ROCKSDB_NAMESPACE::Tickers::TICKER_ENUM_MAX:
        // 0x5F was the max value in the initial copy of tickers to Java.
        // Since these values are exposed directly to Java clients, we keep
        // the value the same forever.
        //
        // TODO: This particular case seems confusing and unnecessary to pin the
        // value since it's meant to be the number of tickers, not an actual
        // ticker value. But we aren't yet in a position to fix it since the
        // number of tickers doesn't fit in the Java representation (jbyte).
        return 0x5F;
```

Reply:
I'm more concerned about if dependent internal code is tied to the ordinal numbers. Can take discussion to https://www.internalfb.com/diff/D42700164

Reply:
Just did some digging in this area while working on deprecating some (other) stats - I'm fairly confident renumbering on the C++ side is safe because it has (organically) happened many times over the years as people added new tickers near related existing ones (as opposed to the end of the list). The Java bindings zag where the core library zigs though.